Pocket/pocket-ios
# refactor(tags): create reusable components and functionality for Add / Edit Tags view

## Summary
Refactor tags list to use tag cell components and set up for Recent Tags to be consistent across both screens

## References 
IN-1268

## Implementation Details
* Separate Add Tags View to three components: `InputTagsView`, `TagsListView`, `TagsCell`
* Utilize `TagsCell` component in the `Edit Tags screen`. This required that the tag types be the same, so created `TagType` that will serve the purpose of stating if the tag is a recent tag, normal tag or not tagged tag. 
* Refactor how we implement the list in terms of arranging tags and created a protocol `TagsList` to be shared across Edit Tags / Add Tags

## Test Steps
Go through the Edit Tags and Add Tags screen to verify that functionality has not changed

## PR Checklist:
- N / A Added Unit / UI tests
